### DISCUSSION TITLE
We've gained access on crates.io to the Android crates again

### DISCUSSION THREAD

**Discussion by madsmtm** [rust-mobile/.github#8] [General]
@kchibisov [managed](https://matrix.to/#/!DGpLzJTRzBDTwZiogk:matrix.org/$d9aXixGIU8t1uhY-xf9MpD8w_K9ynnTLMiZIkwptIH4?via=matrix.org&via=mozilla.org&via=ralith.com) to get access to the Android crates that `@Osspial` was previously the only owner of, how do we want to manage ownership of these?

Should we create sub-teams in @rust-mobile to control this? Who should be the "owner"s of the crates? How do we ensure that the ownership of the crate is not lost again? (At least two people should be owners at all times, preferably three).

Pinging @rust-mobile/rust-mobile-devs.

**Comment by MarijnS95** [comment]
@madsmtm yeah, as already replied I feel like @rib, me, and depending on the project another maintainer/founder should be made owner of the crate. Likewise, we could/should make publishing teams under @rust-mobile, perhaps as you say separated per project.

Finally, it might be desired to "cut ties" with `winit` users and roles if they're not the primary source/contact for crate changes and publishing :)

**Reply by MarijnS95** [reply]
Looking at the list of crates:

- `android_glue`: deprecated and not pushed for 7 years, but still has active downloads. Might want to transfer ownership to one of us to push a `README` with migration guide?
  - Currently only owned by @tomaka
- `ndk`, `ndk-sys`, `ndk-context`: probably 3-4 people that we could make owner:
  - @dvc94ch
  - @msiglreith 
  - @rib 
  - me (@MarijnS95)
- `cargo-apk`, `ndk-build`, `ndk-glue`, `ndk-macro`: Deprecated crates that we might still want to occasionally maintain or at least point to the right source, okay to assign just me and maybe someone else?
  - `cargo-apk` is already owned by me (@MarijnS95)
  - `ndk-macros` and `ndk-context` are already owned by @dvc94ch 
- `xbuild`: Currently owned by @dvc94ch. Could benefit from having one more owner (that would be me, as I've been somewhat actively maintaining the codebase), and be publishable from a team.

**Reply by dvc94ch** [reply]
added @MarijnS95 to xbuild crates

**Comment by kchibisov** [comment]
I've sent invites to @MarijnS95 and @dvc94ch . Feel free to adjust it further on your own.

I've also removed the `rust-windowing/publishing` team access as well as you from this team, since it was only used to work around the case of lacking direct access.

**Reply by MarijnS95** [reply]
On this note I've removed @kchibisov and @osspial from the Android crates on crates.io, and created a new https://github.com/orgs/rust-mobile/teams/publishers team to which we can add anyone who needs publishing rights.

Does anyone have a more specific/granular design choice here? should we use GitHub teams or individuals on crates.io? If we use GitHub teams, should we have segmented teams for the individual groups of crates (one for `ndk`+`ndk-sys`, etc)?

**Comment by kchibisov** [comment]
>android_glue: deprecated and not pushed for 7 years, but still has active downloads. Might want to transfer ownership to one of us to push a README with migration guide?

Do we really need direct access for that though? `rust-windowing/publishing` team have access and can do a release if you really need it? Or what sort of access do you want? I wrote a mail anyway, so we'll see.

**Comment by kchibisov** [comment]
@MarijnS95 @dvc94ch I've sent invite for `android_glue` as well.
